File: CortexFrontend/src/types/cortexAiAssessment.ts
export type CortexAiAssessment = {
  summary: string;
  recommendedPriority: string;
  recommendedStatus: string;
  recommendedCategory: string;
  recommendedOwnerUserId: string | null;
  riskLevel: string;
  confidenceScore: number;
  reasons: string[];
  missingInformation: string[];
  evidence: string[];
};

export function mapAiConfidenceBand(
  confidenceScore: number,
): "High" | "Medium" | "Low" {
  if (confidenceScore >= 0.8) {
    return "High";
  }
  if (confidenceScore >= 0.55) {
    return "Medium";
  }
  return "Low";
}

File: CortexBackend/Models/CortexAiAssessment.cs
namespace Cortex.API.Models;

/// <summary>
/// Unified, vocabulary-constrained AI intake assessment (advisory only; never persisted by the AI layer).
/// </summary>
public sealed class CortexAiAssessment
{
    public string Summary { get; set; } = "";

    public string RecommendedPriority { get; set; } = "";
    public string RecommendedStatus { get; set; } = "";
    public string RecommendedCategory { get; set; } = "";

    public string? RecommendedOwnerUserId { get; set; }

    public string RiskLevel { get; set; } = "";
    public decimal ConfidenceScore { get; set; }

    public List<string> Reasons { get; set; } = [];
    public List<string> MissingInformation { get; set; } = [];

    public List<string> Evidence { get; set; } = [];
}
